src/gui/instrument/EnvelopeGraph.cpp

```diff
@@ -54,102 +56,208 @@ void EnvelopeGraph::modelChanged()
 
 void EnvelopeGraph::mousePressEvent(QMouseEvent* me)
 {
-	if(me->button() == Qt::LeftButton)
+	if (me->button() == Qt::LeftButton) { toggleAmountModel(); }
+}
+
+void EnvelopeGraph::contextMenuEvent(QContextMenuEvent* event)
+{
+	QMenu menu(this);
+	QMenu* scalingMenu = menu.addMenu(tr("Scaling"));
+	scalingMenu->setToolTipsVisible(true);
+
+	auto switchTo = [&](ScalingMode scaling)
 	{
-		toggleAmountModel();
-	}
+		if (m_scaling != scaling)
+		{
+			m_scaling = scaling;
+			update();
+		}
```

Comment:
Just opinionated styling here... Less indention :)
```suggestion
		if (m_scaling == scaling) { return; }
		m_scaling = scaling;
		update();
```

Reply:
I have rather kept this as is because to me the current solution communicates better that something happens only if the condition holds.

Reply:
I'd say this still communicates that?

Reply:
Yes, it also communicates that but as I've written "to me" the other version communicates it better. And as you've already written it was an "opinionated styling" proposal anyway. :wink: